test/cli/utils_test.go

```diff
@@ -271,7 +274,137 @@ func getSyftBinaryLocation(t testing.TB) string {
 		// SYFT_BINARY_LOCATION is the absolute path to the snapshot binary
 		return os.Getenv("SYFT_BINARY_LOCATION")
 	}
-	return getSyftBinaryLocationByOS(t, runtime.GOOS)
+	bin := getSyftBinaryLocationByOS(t, runtime.GOOS)
+
+	// only run on valid bin target, when not running in CI
+	if bin != "" && os.Getenv("CI") == "" {
+		buildSyft(t, bin)
+		// regardless if we have a successful build, don't attempt to keep building
+		_ = os.Setenv("SYFT_BINARY_LOCATION", bin)
+	}
+
+	return bin
+}
+
+func buildSyft(t testing.TB, outfile string) {
+	dir := repoRoot(t)
+
+	start := time.Now()
+
+	var stdout, stderr string
+	var err error
+	switch "go" {
```

Comment:
Change this to try the different variants.

Reply:
I like the idea of by default having the goreleaser approach done since it would be most consistent with how we're going to release. That being said, I also like that this is swappable strategy... we could simply leave this here and wire it up to an environment variable so the dev could choose between go and goreleaser (should probably removed the make approach though).